lib/paper_trail/frameworks/active_record/models/paper_trail/version.rb

```diff
@@ -7,5 +7,12 @@ module PaperTrail
   # See the readme for details.
   class Version < ::ActiveRecord::Base
     include PaperTrail::VersionConcern
+
+    # You can now get user object (supports Device defaults)
+    def user(user_model = :User, primary_key = :id)
+      unless whodunnit.nil?
+      	user_model.to_s.constantize.find_by(primary_key.to_s.concat(' = ?'), whodunnit)
+      end
+    end
```

Comment:
I like the idea of taking the class name as an argument, and I like the default, `:User`.

The second argument, `primary_key`, is unnecessary. We should use `ActiveResource::Base#primary_key`.

The code should live in `version_concern.rb`. We don't put any code in `version.rb`.

Finally, I believe the hash form of `find_by` is preferable.

Putting that all together:

```ruby
def user(user_model = :User)
  unless whodunnit.nil?
    model_class = user_model.to_s.constantize
    model_class.find_by(model_class.primary_key => whodunnit)
  end
end
```